modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/internal/freemarker/tool/java/parser/ResourceOpenAPIParser.java

```diff
@@ -1058,8 +1058,27 @@ else if (pathSegment.contains("{")) {
 				String previousMethodNameSegment = operationIdSegments.get(
 					operationIdSegments.size() - 1);
 
-				if (!previousMethodNameSegment.endsWith(pathName) &&
-					!previousMethodNameSegment.endsWith(schemaName)) {
+				if (pathName.endsWith("ExternalReferenceCode") &&
+					ConfigUtil.isVersionCompatible(configYAML, 10)) {
+
+					String externalReferenceCodeSubjectName =
+						StringUtil.upperCaseFirstLetter(
+							CamelCaseUtil.toCamelCase(
+								pathSegment.replaceAll(
+									"\\{|-?ExternalReferenceCode}", "")));
+
+					if (!(previousMethodNameSegment.endsWith(
```

Comment:
I need more time to analyze this. I am finishing this review and will do it in the following one